code/modules/uplink/uplink.dm

```diff
@@ -74,25 +74,25 @@ GLOBAL_LIST_EMPTY(uplinks)
 		return	//no hitting everyone/everything just to try to slot tcs in!
 	if(istype(I, /obj/item/stack/telecrystal))
 		LoadTC(user, I)
-	for(var/item in subtypesof(/datum/uplink_item))
-		var/datum/uplink_item/UI = item
-		var/path = null
-		if(initial(UI.refund_path))
-			path = initial(UI.refund_path)
-		else
-			path = initial(UI.item)
-		var/cost = 0
-		if(initial(UI.refund_amount))
-			cost = initial(UI.refund_amount)
-		else
-			cost = initial(UI.cost)
-		var/refundable = initial(UI.refundable)
-		if(I.type == path && refundable && I.check_uplink_validity())
-			telecrystals += cost
-			purchase_log.total_spent -= cost
-			to_chat(user, "<span class='notice'>[I] refunded.</span>")
-			qdel(I)
-			return
+	for(var/category in uplink_items)
+		for(var/item in uplink_items[category])
+			var/datum/uplink_item/UI = uplink_items[category][item]
+			var/path = null

```

Comment:
These ifs could probably be compressed, `var/path = UI.refund_path || UI.item` and likewise for the other one.